test/core/end2end/tests/compressed_payload.c

```diff
@@ -329,6 +330,18 @@ static void request_with_payload_template(
   server_args = grpc_channel_args_set_compression_algorithm(
       NULL, default_server_channel_compression_algorithm);
 
+  if (user_agent_override) {
```

Comment:
Did it this way because a fixture only sets one user-agent string, while I need positive/negative tests here. It is a bit ugly indeed... I think I should put this in a separate test case.